lib-injection/host_inject.rb

```diff
@@ -0,0 +1,97 @@
+return if ENV["DD_TRACE_SKIP_LIB_INJECTION"] == "true"
+
+Gem.paths = {
+  "GEM_PATH" => "/opt/datadog/apm/library/ruby/#{RUBY_VERSION.split(".")[0..1].join(".")}:/opt/datadog/apm/library/ruby:#{ENV["GEM_PATH"]}"
+}
+
+ENV["GEM_PATH"] = Gem.path.join(":")
+
+begin
+  require "open3"
+  require "bundler"
+  require "shellwords"
+  require "fileutils"
+
+  failure_prefix = "Datadog lib injection failed:"
+  support_message = "For help solving this issue, please contact Datadog support at https://docs.datadoghq.com/help/."
+
+  unless Bundler::SharedHelpers.in_bundle?
+    STDOUT.puts "[ddtrace] Not in bundle... skipping host injection" if ENV["DD_TRACE_DEBUG"] == "true"
```

Comment:
```suggestion
    debug_log "[ddtrace] Not in bundle... skipping host injection"
```

I recommend extracting debug logging to a method, as it is used multiple times in this file.